langserver/format.go

```diff
@@ -28,23 +29,34 @@ func (h *LangHandler) handleTextDocumentFormatting(ctx context.Context, conn jso
 	}
 
 	filename := h.FilePath(params.TextDocument.URI)
-	bctx := h.BuildContext(ctx)
-	fset := token.NewFileSet()
-	file, err := buildutil.ParseFile(fset, bctx, nil, path.Dir(filename), path.Base(filename), parser.ParseComments)
-	if err != nil {
-		return nil, err
-	}
 
-	ast.SortImports(fset, file)
+	var formatted []byte
+	if h.config.GoimportsEnabled {
+		imports.LocalPrefix = h.config.GoimportsLocalPrefix
+		var err error
+		formatted, err = imports.Process(filename, nil, nil)
```

Comment:
The go-langserver maintains its own file cache, like all other LSP servers do.
So by passing `nil` as second parameter to `Process` forces the imports code to actually read the file from disk without trying to read from the internal file cache!
Just move the `unformatted` variable before this `if` and pass the `unformatted` as second argument to `Process`